HIPs/firefly.md

```diff
@@ -0,0 +1,374 @@
+
+Hyperledger FireFly brings together (1), (2), (3), and (4) to offer a pre-integrated set of runtimes via a highly pluggable framework. Note that FireFly is NOT attempting to replace or invalidate any specific runtime technology, but rather pulls together multiple complementary technologies into a coherent, larger system. Indeed, many of the infrastructure runtimes FireFly relies on are separate open source projects.
+
+Enterprise blockchains are purpose built to deliver (1) with invaluable attributes like _global ordering_ and _finality_. FireFly will support Hyperledger Fabric, Enterprise Ethereum (such as Quorum and Besu), and will investigate Corda support. (1) also includes global storage providers such as IPFS. FireFly also provides a significant amount of technology for (2) and (3) including cross-party messaging, cross-party private document transfer, and event propagation. All of these runtimes and connectors into the FireFly node. For (4), Hyperledger FireFly also provides a network map and registry that unifies all of the identity and bindings information needed.
+
+## Motivation
+
+When building a system for multiparty business transactions based on blockchain technology, there are a number of problems that need to be solved. Problems such as:
+
+- How do parties agree on standardized data types?
+- How do I securely send private data between two parties in the system without other parties being able to view the data?
+- How do I run business processes in response to blockchain transactions?
+- How do I leverage blockchain technology, but not expose sensitive data stored on-chain?
+- How do I process events originated from multiple parties in the correct order that is uniformly honored throughout the decentralized system?
+
+Today, there is no comprehensive OSS project that solves all of these problems. FireFly fills this gap, and allows businesses to focus on solving business problems.
+
+![Typical Blockchain Stack](../assets/stack.svg "Typical Blockchain Stack")
+
+The gap includes functionality such as registering and verifying member identity within the network, defining custom data types, sharing private or public data with network members, hashing and pinning data to the blockchain, securely sending events to network members, and triggering business process flows in response to events.
+
+Because there is no existing OSS project that solves these problems, developers are left to themselves to create a solution. These solutions likely end up becoming specialized or tightly coupled to the project they were designed as a part of, and therefore are not reusable. Firefly seeks to fill this gap with a platform that is reusable for any multiparty business transaction backed by blockchain.
+
+## Status
+
+Proposal
+
+## Solution
```

Comment:
I must be missing something but where is in this Section the descriptions of the solutions to problems (1), (2), (3), (4), and those in the motivation section?